depends/packages/qt.mk

```diff
@@ -1,122 +1,104 @@
 $(package)_patches += qtbase-moc-ignore-gcc-macro.patch
-$(package)_patches += no_warnings_for_symbols.patch
+$(package)_patches += qtbase_avoid_native_float16.patch
+$(package)_patches += qtbase_avoid_qmain.patch
 $(package)_patches += rcc_hardcode_timestamp.patch
-$(package)_patches += duplicate_lcqpafonts.patch
-$(package)_patches += guix_cross_lib_path.patch
-$(package)_patches += fix-macos-linker.patch
-$(package)_patches += memory_resource.patch
-$(package)_patches += clang_18_libpng.patch
 $(package)_patches += utc_from_string_no_optimize.patch
 $(package)_patches += windows_lto.patch
-$(package)_patches += darwin_no_libm.patch
-$(package)_patches += zlib-timebits64.patch
+$(package)_patches += macos_skip_version_checks.patch
+$(package)_patches += guard_headers_properly.patch
+$(package)_patches += qttools_skip_dependencies.patch
 
-$(package)_qttranslations_file_name=qttranslations-$($(package)_suffix)
-$(package)_qttranslations_sha256_hash=5b94d1a11b566908622fcca2f8b799744d2f8a68da20be4caa5953ed63b10489
+$(package)_qttranslations_file_name=$(qt_details_qttranslations_file_name)
+$(package)_qttranslations_sha256_hash=$(qt_details_qttranslations_sha256_hash)
 
-$(package)_qttools_file_name=qttools-$($(package)_suffix)
-$(package)_qttools_sha256_hash=12061a85baf5f4de8fbc795e1d3872b706f340211b9e70962caeffc6f5e89563
+$(package)_qttools_file_name=$(qt_details_qttools_file_name)
+$(package)_qttools_sha256_hash=$(qt_details_qttools_sha256_hash)
 
 $(package)_extra_sources  = $($(package)_qttranslations_file_name)
 $(package)_extra_sources += $($(package)_qttools_file_name)
 
+$(package)_top_download_path=$(qt_details_top_download_path)
+$(package)_top_cmakelists_file_name=$(qt_details_top_cmakelists_file_name)
+$(package)_top_cmakelists_download_file=$(qt_details_top_cmakelists_download_file)
+$(package)_top_cmakelists_sha256_hash=$(qt_details_top_cmakelists_sha256_hash)
+$(package)_top_cmake_download_path=$(qt_details_top_cmake_download_path)
+$(package)_top_cmake_ecmoptionaladdsubdirectory_file_name=$(qt_details_top_cmake_ecmoptionaladdsubdirectory_file_name)
+$(package)_top_cmake_ecmoptionaladdsubdirectory_download_file=$(qt_details_top_cmake_ecmoptionaladdsubdirectory_download_file)
+$(package)_top_cmake_ecmoptionaladdsubdirectory_sha256_hash=$(qt_details_top_cmake_ecmoptionaladdsubdirectory_sha256_hash)
+$(package)_top_cmake_qttoplevelhelpers_file_name=$(qt_details_top_cmake_qttoplevelhelpers_file_name)
+$(package)_top_cmake_qttoplevelhelpers_download_file=$(qt_details_top_cmake_qttoplevelhelpers_download_file)
+$(package)_top_cmake_qttoplevelhelpers_sha256_hash=$(qt_details_top_cmake_qttoplevelhelpers_sha256_hash)
+
+$(package)_extra_sources += $($(package)_top_cmakelists_file_name)
+$(package)_extra_sources += $($(package)_top_cmake_ecmoptionaladdsubdirectory_file_name)
+$(package)_extra_sources += $($(package)_top_cmake_qttoplevelhelpers_file_name)
+
 define $(package)_set_vars
-$(package)_config_env = QT_MAC_SDK_NO_VERSION_CHECK=1
```

Comment:
Looks like this was dropped entirely. Any particualr reason? It was previously was suppressing:
```bash
CMake Warning at qtbase/cmake/QtPublicAppleHelpers.cmake:920 (message):
  Qt requires at least version 14 of Xcode, you're building against version .
  Please upgrade.  You can turn this warning into an error by configuring
  with -DQT_FORCE_FATAL_APPLE_SDK_AND_XCODE_CHECK=ON.
Call Stack (most recent call first):
  qtbase/cmake/QtBuildHelpers.cmake:443 (_qt_internal_check_apple_sdk_and_xcode_versions)
  qtbase/cmake/QtBuild.cmake:4 (qt_internal_setup_build_and_global_variables)
  qtbase/cmake/QtSetup.cmake:6 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:21 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:179 (qt_build_internals_set_up_private_api)
  qtbase/cmake/QtBaseHelpers.cmake:154 (qt_build_repo_begin)
  qtbase/CMakeLists.txt:32 (qt_internal_qtbase_build_repo)


CMake Warning at qtbase/cmake/QtPublicAppleHelpers.cmake:934 (message):
  Qt has only been tested with version 14 of the platform SDK, you're using
  15.0.  This is an unsupported configuration.  You may experience build
  issues, and by using the 15.0 SDK you are opting in to new features that Qt
  has not been prepared for.  Please downgrade the SDK you use to build your
  app to version 14, or configure with -DQT_NO_APPLE_SDK_MAX_VERSION_CHECK=ON
  to silence this warning.
Call Stack (most recent call first):
  qtbase/cmake/QtBuildHelpers.cmake:443 (_qt_internal_check_apple_sdk_and_xcode_versions)
  qtbase/cmake/QtBuild.cmake:4 (qt_internal_setup_build_and_global_variables)
  qtbase/cmake/QtSetup.cmake:6 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:21 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:179 (qt_build_internals_set_up_private_api)
  qtbase/cmake/QtBaseHelpers.cmake:154 (qt_build_repo_begin)
  qtbase/CMakeLists.txt:32 (qt_internal_qtbase_build_repo)
```

Athough this may also hint at some other issues, given it seems Qts SDK version detection isn't working properly here? i.e `you're building against against version .`

Reply:
Addressed in https://github.com/bitcoin/bitcoin/pull/30997#issuecomment-2402990346.

Reply:
> Addressed in https://github.com/bitcoin/bitcoin/pull/30997#issuecomment-2402990346.

How was it addressed? I still see the same output using 2d009aa8e861c161820a4c0619ecd4b2329267af:
```bash
make -C depends -j12 HOST=x86_64-apple-darwin
gmake: Entering directory '/bitcoin/depends'
Fetching qtbase-everywhere-src-6.7.3.tar.xz from https://download.qt.io/official_releases/qt/6.7/6.7.3/submodules
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
<snip>
-- Configuring submodule 'qtbase'
-- [QtBase] CMAKE_BUILD_TYPE was already explicitly set to: 'Release'
CMake Warning at qtbase/cmake/QtPublicAppleHelpers.cmake:920 (message):
  Qt requires at least version 14 of Xcode, you're building against version .
  Please upgrade.  You can turn this warning into an error by configuring
  with -DQT_FORCE_FATAL_APPLE_SDK_AND_XCODE_CHECK=ON.
Call Stack (most recent call first):
  qtbase/cmake/QtBuildHelpers.cmake:443 (_qt_internal_check_apple_sdk_and_xcode_versions)
  qtbase/cmake/QtBuild.cmake:4 (qt_internal_setup_build_and_global_variables)
  qtbase/cmake/QtSetup.cmake:6 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:21 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:179 (qt_build_internals_set_up_private_api)
  qtbase/cmake/QtBaseHelpers.cmake:154 (qt_build_repo_begin)
  qtbase/CMakeLists.txt:32 (qt_internal_qtbase_build_repo)


CMake Warning at qtbase/cmake/QtPublicAppleHelpers.cmake:934 (message):
  Qt has only been tested with version 14 of the platform SDK, you're using
  15.0.  This is an unsupported configuration.  You may experience build
  issues, and by using the 15.0 SDK you are opting in to new features that Qt
  has not been prepared for.  Please downgrade the SDK you use to build your
  app to version 14, or configure with -DQT_NO_APPLE_SDK_MAX_VERSION_CHECK=ON
  to silence this warning.
Call Stack (most recent call first):
  qtbase/cmake/QtBuildHelpers.cmake:443 (_qt_internal_check_apple_sdk_and_xcode_versions)
  qtbase/cmake/QtBuild.cmake:4 (qt_internal_setup_build_and_global_variables)
  qtbase/cmake/QtSetup.cmake:6 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:21 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:179 (qt_build_internals_set_up_private_api)
  qtbase/cmake/QtBaseHelpers.cmake:154 (qt_build_repo_begin)
  qtbase/CMakeLists.txt:32 (qt_internal_qtbase_build_repo)


-- [QtBase] Checking for feature set changes
-- [QtBase] The OBJC compiler identification is AppleClang 16.0.0.16000026
```

Reply:
> How was it addressed? I still see the same output using...

I assume you're still observing the warning for the `native_qt` package, not the `qt` package. In that case, I don’t think we need to worry about it.